pkg/resources/stream_on_external_table.go

```diff
@@ -0,0 +1,223 @@
+package resources
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/internal/provider"
+
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/helpers"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/customdiff"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+var streamOnExternalTableSchema = func() map[string]*schema.Schema {
+	streamOnExternalTable := map[string]*schema.Schema{
+		"external_table": {
+			Type:             schema.TypeString,
+			Required:         true,
+			Description:      blocklistedCharactersFieldDescription("Specifies an identifier for the external table the stream will monitor."),
+			DiffSuppressFunc: SuppressIfAny(suppressIdentifierQuoting, IgnoreChangeToCurrentSnowflakeValueInShow("table_name")),
+			ValidateDiagFunc: IsValidIdentifier[sdk.SchemaObjectIdentifier](),
+		},
+		"insert_only": {
+			Type:             schema.TypeString,
+			Optional:         true,
+			Default:          BooleanDefault,
+			ValidateDiagFunc: validateBooleanString,
+			DiffSuppressFunc: IgnoreChangeToCurrentSnowflakeValueInShowWithMapping("mode", func(x any) any {
+				return x.(string) == string(sdk.StreamModeInsertOnly)
+			}),
+			Description: booleanStringFieldDescription("Specifies whether this is an insert-only stream."),
+		},
+		AtAttributeName:     atSchema,
+		BeforeAttributeName: beforeSchema,
+	}
+	return helpers.MergeMaps(streamCommonSchema, streamOnExternalTable)
+}()
+
+func StreamOnExternalTable() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: CreateStreamOnExternalTable(false),
+		ReadContext:   ReadStreamOnExternalTable(true),
+		UpdateContext: UpdateStreamOnExternalTable,
+		DeleteContext: DeleteStreamContext,
+		Description:   "Resource used to manage streams on external tables. For more information, check [stream documentation](https://docs.snowflake.com/en/sql-reference/sql/create-stream).",
+
+		CustomizeDiff: customdiff.All(
+			ComputedIfAnyAttributeChanged(streamOnExternalTableSchema, ShowOutputAttributeName, "external_table", "insert_only", "comment"),
+			ComputedIfAnyAttributeChanged(streamOnExternalTableSchema, DescribeOutputAttributeName, "external_table", "insert_only", "comment"),
+		),
+
+		Schema: streamOnExternalTableSchema,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: ImportStreamOnExternalTable,
+		},
+	}
+}
+
+func ImportStreamOnExternalTable(ctx context.Context, d *schema.ResourceData, meta any) ([]*schema.ResourceData, error) {
+	log.Printf("[DEBUG] Starting stream import")
+	client := meta.(*provider.Context).Client
+	id, err := sdk.ParseSchemaObjectIdentifier(d.Id())
+	if err != nil {
+		return nil, err
+	}
+
+	v, err := client.Streams.ShowByID(ctx, id)
+	if err != nil {
+		return nil, err
+	}
+	if err := d.Set("name", id.Name()); err != nil {
```

Comment:
nit: these could be extracted because they set a few common stream fields

Reply:
I'll do this in the next PR.